docs/ACCESSIBILITY.md

```diff
@@ -0,0 +1,139 @@
+-  It is optional to allow <kbd>TAB</kbd> key to focus on this element
+
+### Tabbable
+
+-  A [focusable](#focusable) which can be focused on by pressing <kbd>TAB</kbd> key
+
+## UX: A series of decisions
+
+> This is related to https://github.com/microsoft/BotFramework-WebChat/issues/3135.
+
+### User story
+
+The bot send a question with a set of predefined answers as UI buttons that will drive the conversation towards a particular goal (a.k.a. decision buttons).
+
+After the user make the decision by clicking on a button, the decision is being submitted. The user is not allowed to reselect another decision.
+
+The bot will then send another question with another set of answers.
+
+### Positive user experience
+
+We should disable buttons that are not part of the decision. Momentarily, the next set of answers will arrive. Since we must not change focus asynchronously outside of user gesture, the user is required to press <kbd>TAB</kbd> to get into the next set of decision buttons.
+
+When the user <kbd>TAB</kbd> out of the current button and into the next set of buttons, we should disable all previous decision buttons including the one the user chose. This will give a more consistent UX on how buttons are disabled.
+
+Since disabling buttons will also hide them from screen reader, we should add a screen reader-only text to tell the user which answer they chose.
+
+### Exceptions
+
+-  If no tabbable UI after the current disabled button, the next <kbd>TAB</kbd> should goes into the send box.
+
+### Implementation
+
+When an UI is being disabled:
+
+-  All UI will be manually styled, based on `:disabled, [aria-disabled="true"]` query
+   -  User agent stylesheet do not take account into `aria-disabled` attribute
+-  Set `aria-disabled` attribute to `true`
+   -  If it is a `<button>`, set `onClick` to a handler that call `event.preventDefault()`
+   -  If it is a `<input type="text">` or `<textarea>`, set `readOnly` to `true`
+-  If it is currently focused, wait until `onBlur` event and set `disabled` attribute
+   -  Otherwise, set `disabled` attribute immediately
+
+> List of elements support `disabled` attribute can be found in [this article](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled).
+
+### Additional context
+
+By default, HTML is static. Thus, the default `disabled` implementation works on a static web page.
+
+On a dynamic web page, when `disabled` is being applied to a focusing element (`document.activeElement`), the focus change varies between browsers:
+
+-  Chrome will keep the focus on current element invisible to JavaScript or CSS
+   -  On <kbd>TAB</kbd>, it jump to next tabbable sibling or descendants of them (depth-first search)
+-  Firefox will send the focus to the parent, invisibly
+   -  On <kbd>TAB</kbd>, it jump to the first tabbable sibling or descendants of them (depth-first search)
+-  Edge UWP and IE11 will send the focus to `<body>`, invisibly
```

Comment:
Again, not sure what you mean by 'invisibly.' Is it invisible to js and css like with Chrome?

Reply:
Changed to table format.